compact_str/src/repr/mod.rs

```diff
@@ -381,6 +392,10 @@ impl Repr {
     #[allow(clippy::len_without_is_empty)] // is_empty exists on CompactString
     #[inline]
     pub fn len(&self) -> usize {
+        if let Some(s) = self.as_static_str() {
+            return s.len();
+        }
```

Comment:
see comment on `fn as_slice(...)`

Reply:
Thanks, I've applied the same optimization here.